client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala

```diff
@@ -434,48 +437,55 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
       reduceId: Int,
       oldEpochId: Int,
       oldPartition: PartitionLocation): Unit = {
+
+    def reply(response: ChangeLocationResponse): Unit = {
+      contexts.synchronized {
+        contexts.remove(reduceId)
+      }.asScala.foreach(_.reply(response))
+    }
+
     val candidates = workersNotBlacklisted(shuffleId)
-    val slots = if (oldPartition != null) {
+    if (candidates.size < 1 || (ShouldReplicate && candidates.size < 2)) {
+      logError("[Update partition] failed for not enough candidates for revive.")
+      reply(ChangeLocationResponse(StatusCode.SlotNotAvailable, null))
+      return null
+    }
+
+    val revivedLocation = if (oldPartition != null) {
```

Comment:
This slot can be revived or split.  There is no reason to assume this slot is intended for reviving context.